contracts/TokenConverter/RiskFundConverter.sol

```diff
@@ -151,42 +151,6 @@ contract RiskFundConverter is AbstractTokenConverter {
         return poolsAssetsReserves[comptroller][asset];
     }
 
-    /// @dev Update the reserve of the asset for the specific pool after transferring to risk fund
-    /// and transferring funds to the protocol share reserve
-    /// @param comptroller Comptroller address (pool)
-    /// @param asset Asset address
-    /// @custom:event AssetTransferredToDestination emits when poolsAssetsDirectTransfer is enabled for entered comptroller and asset
-    /// @custom:event AssetsReservesUpdated emits when poolsAssetsDirectTransfer is not enabled for entered comptroller and asset
-    function updateAssetsState(address comptroller, address asset) public nonReentrant {
-        if (!ensureAssetListed(comptroller, asset)) revert MarketNotExistInPool(comptroller, asset);
-
-        IERC20Upgradeable token = IERC20Upgradeable(asset);
-        uint256 currentBalance = token.balanceOf(address(this));
-        uint256 assetReserve = assetsReserves[asset];
-        if (currentBalance > assetReserve) {
-            uint256 balanceDifference;
-            unchecked {
-                balanceDifference = currentBalance - assetReserve;
-            }
-            if (poolsAssetsDirectTransfer[comptroller][asset]) {
-                uint256 previousDestinationBalance = token.balanceOf(destinationAddress);
-                token.safeTransfer(destinationAddress, balanceDifference);
-                uint256 newDestinationBalance = token.balanceOf(destinationAddress);
-
-                emit AssetTransferredToDestination(comptroller, asset, balanceDifference);
-                IRiskFund(destinationAddress).updatePoolState(
-                    comptroller,
-                    asset,
-                    newDestinationBalance - previousDestinationBalance
-                );
-            } else {
-                assetsReserves[asset] += balanceDifference;
-                poolsAssetsReserves[comptroller][asset] += balanceDifference;
-                emit AssetsReservesUpdated(comptroller, asset, balanceDifference);
-            }
```

Comment:
Why don't we have this block in the new `_updateAssetsState` function?

Reply:
First, we check for the `directTransfer` in the `_updateAssetsState` and then do `_privateConversion` after the conversion we update the states of the converter according to the tokens converted or directly transferred.